exp/api/v1beta1/azuremachinepool_default.go

```diff
@@ -57,3 +57,14 @@ func (amp *AzureMachinePool) SetSpotEvictionPolicyDefaults() {
 		amp.Spec.Template.SpotVMOptions.EvictionPolicy = &defaultPolicy
 	}
 }
+
+// SetDiagnosticsDefaults sets the defaults for Diagnostic settings for an AzureMachinePool.
+func (amp *AzureMachinePool) SetDiagnosticsDefaults() {
+	if amp.Spec.Template.Diagnostics == nil {
```

Comment:
Currently the API makes `Diagnostics.Boot` optional which would allow the user to set 

```
template:
    diagnostics: {}
``` 

this would cause the default to not get applied and result in an invalid configuration. We should either make `Boot` required (ie. remove the //+optional tag) when Diagnostics is specified or also apply defaults in here when `amp.Spec.Template.Diagnostics` != nil